Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -229,21 +230,63 @@ public IObservable<Item> ListObjectsAsync(ListObjectsArgs args, CancellationToke
+                        if (objectList is null) return;
+                        // Add user metadata information
+                        if (goArgs.IncludeUserMetadata)
+                        {
+                            var indx = 0;
+                            foreach (var itm in objectList.Item2)
+                            {
+                                var statObjectArgs = new StatObjectArgs()
+                                    .WithBucket(args.BucketName)
+                                    .WithObject(itm.Key);
+                                var objStat = await StatObjectAsync(statObjectArgs, cancellationToken)
+                                    .ConfigureAwait(false);
+                                objectList.Item1.UserMetadata = [];
+                                itm.UserMetadata = [];
+                                foreach (var pair in objStat.MetaData)
+                                {
+                                    objectList.Item1.UserMetadata.Add(new MetadataItem(pair.Key, pair.Value));
+                                    itm.UserMetadata.Add(new MetadataItem(pair.Key, pair.Value));
+                                }
+
+                                indx++;
+                            }
+                        }
+
+                        var listObjectsItemResponse = new ListObjectVersionResponse(objectList, obs);
                         if (objectList.Item2.Count == 0 && count == 0) return;
 
                         obs = listObjectsItemResponse.ItemObservable;
-                        marker = listObjectsItemResponse.NextKeyMarker;
                         versionIdMarker = listObjectsItemResponse.NextVerMarker;
                         isRunning = objectList.Item1.IsTruncated;
                     }
                     else
                     {
                         var objectList = await GetObjectListAsync(goArgs, cts.Token).ConfigureAwait(false);
-                        if (objectList.Item2.Count == 0 &&
-                            objectList.Item1.KeyCount.Equals("0", StringComparison.OrdinalIgnoreCase) && count == 0)
+                        if (objectList is null ||
+                            (objectList.Item2.Count == 0 &&
+                             objectList.Item1.KeyCount.Equals("0", StringComparison.Ordinal) &&
+                             count == 0))
                             return;
 
+                        // Add user metadata information
+                        if (goArgs.IncludeUserMetadata)
+                        {
+                            var indx = 0;
+                            foreach (var itm in objectList.Item2)
+                            {
+                                var statObjectArgs = new StatObjectArgs()
+                                    .WithBucket(args.BucketName)
+                                    .WithObject(itm.Key);
+                                var objStat = await StatObjectAsync(statObjectArgs, cancellationToken)
```

Comment:
See the previous comment that we should use the `metadata=true` solution that is also used in the Go SDK.

Reply:
The support decision for this feature is made for all MinIO SDKs.
So, I don't think this decision will be changed IMHO, but there is no harm discussing it. 
This will be discussed with @abperiasamy  and @kannappanr anyways.

Reply:
The problem is not the feature itself, it is the way it is implemented.

Reply:
Yes, I understand.
Working on it.